reflex/app.py

```diff
@@ -659,81 +654,119 @@ def compile(self):
             # If a theme component was provided, wrap the app with it
             app_wrappers[(20, "Theme")] = self.theme
 
-        with progress:
-            for route, component in self.pages.items():
-                # TODO: this progress does not reflect actual threaded task completion
+        with progress, concurrent.futures.ThreadPoolExecutor() as thread_pool:
+            fixed_pages = 7
+            compile_total = sum(
+                [
+                    len(self.pages),  # Memoize stateful components
+                    len(self.pages) + fixed_pages,  # Compile pages
+                ]
+            )
+            task = progress.add_task("Memoize:", total=compile_total)
+
+            def mark_complete(_=None):
```

Comment:
Can we combine this whole step into just one on the progress bar. Reducing the number on the steps makes the framework seem more lightweight, plus each of these steps is pretty instant on my machine